### DISCUSSION TITLE
Notifications always disappear after 5 seconds - Linux & Dunst

### DISCUSSION THREAD

**Discussion by llitz** [wireapp/wire-desktop#5083] [Ideas]
## Wire for web version
2020-07-24-13-46

## Browser version
Chromium 84.0.4147.89
Firefox 78.0.2

## Installed browser addons
Disabled all of them while wire is running

## Operating system
Linux - Gentoo (should be replaceable on any distribution)
Relying on dunst for messages (1.4.1)

## What steps will reproduce the problem?
Receive a message on wire webapp not currently focused.
Chrome (configured for desktop notifications) and firefox will use dbus to send a notification
After 5 seconds, Wire sends a command to close notification
Browser follows up and the notification is closed

## What is the expected result?
Notification should only be removed when the message is read, either in another device or on the browser tab.

## What is the actual result?
Browser still has an unread notification but is forcefully removing the notification from system-tray.

Apparently we have it hard-coded to timeout a notification after 5 seconds.
This is confirmed via a dbus message for NotificationClosed.
Perhaps it would be better to attach the removal of this notification to when we update the chat conversation symbol for unread messages for that person.

**Comment by ffflorian** [comment]
Hi @llitz, unfortunately we don't have the resources to support other notification daemons, so I am adding a Feature Request label and if we have the resources in the future we can look into it.

**Comment by llitz** [comment]
Hi Florian,

I don't think I have explained properly, but this is isn't about different
daemon, but the default notification system on the webapp.

At least on Linux, the notifications on webapps, the one where we display
the messages content, are being closed after 5 seconds.

Checking the code, this 5 seconds rule seems to be hard-coded.

We already have code that will add a small red circle to the application
icon's in the browser tab when there's an unread message and remove it
after the message is read.

What I am suggesting is that we remove the 5 seconds rules to clear message
notification and send the command to clear it at the same when we are
clearing the notification icon (or something similar to that).

If you look at WhatsApp and a few other apps, they have a counter instead
of a red dot and messages are removed as the counter of unread messages
decrease.

**Comment by ffflorian** [comment]
> What I am suggesting is that we remove the 5 seconds rules to clear message
> notification and send the command to clear it at the same when we are
> clearing the notification icon (or something similar to that).

Ah, I understand! Alright, I'll forward it to the team to see if we can change that.
